bokehjs/src/coffee/common/selector.coffee

```diff
@@ -7,25 +7,77 @@ p = require "../core/properties"
 class Selector extends HasProps
   type: 'Selector'
 
-  update: (indices, final, append, silent=false) ->
+  select: (tool, renderer_view, geometry, final=true) ->
+    if @source != renderer_view.model.data_source
+      logger.warn('select called with mis-matched data sources')
```

Comment:
I think this issue will/would be obviated by resolved the goal described [below](https://github.com/bokeh/bokeh/pull/5158#issuecomment-247088450)

If you have any suggestions, that'd be great.